File: CS/WinSolution.Module.Win/SchedulerListViewViewController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Scheduler.Win;
using DevExpress.XtraScheduler;

namespace WinSolution.Module.Win {
    public class SchedulerListViewViewController : ViewController<ListView> {
        private SchedulerListEditor listEditor;
        public SchedulerListViewViewController() {
            TargetObjectType = typeof(DevExpress.Persistent.BaseImpl.Event);
        }
        protected override void OnViewControlsCreated() {
            base.OnViewControlsCreated();
            listEditor = View.Editor as SchedulerListEditor;
            if (listEditor != null) {
                SchedulerControl scheduler = (SchedulerControl)listEditor.Scheduler;
                scheduler.InitAppointmentDisplayText -= scheduler_InitAppointmentDisplayText;
                scheduler.InitAppointmentDisplayText += scheduler_InitAppointmentDisplayText;
            }
        }
        protected override void OnDeactivating() {
            base.OnDeactivating();
            if (listEditor != null) {
                SchedulerControl scheduler = (SchedulerControl)listEditor.Scheduler;
                scheduler.InitAppointmentDisplayText -= scheduler_InitAppointmentDisplayText;
            }
        }
        private void scheduler_InitAppointmentDisplayText(object sender, AppointmentDisplayTextEventArgs e) {
            Appointment appointment = e.Appointment;
            DevExpress.Persistent.BaseImpl.Event obj = (DevExpress.Persistent.BaseImpl.Event)listEditor.SourceObjectHelper.GetSourceObject(appointment);
            e.Text = string.Format("{0}-({1})", e.Text, obj.GetHashCode());
        }
    }
}

File: VB/WinSolution.Module.Win/SchedulerListViewViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Scheduler.Win
Imports DevExpress.XtraScheduler

Namespace WinSolution.Module.Win
	Public Class SchedulerListViewViewController
		Inherits ViewController(Of ListView)
		Private listEditor As SchedulerListEditor
		Public Sub New()
			TargetObjectType = GetType(DevExpress.Persistent.BaseImpl.Event)
		End Sub
		Protected Overrides Overloads Sub OnViewControlsCreated()
			MyBase.OnViewControlsCreated()
			listEditor = TryCast(View.Editor, SchedulerListEditor)
			If listEditor IsNot Nothing Then
				Dim scheduler As SchedulerControl = CType(listEditor.Scheduler, SchedulerControl)
				RemoveHandler scheduler.InitAppointmentDisplayText, AddressOf scheduler_InitAppointmentDisplayText
				AddHandler scheduler.InitAppointmentDisplayText, AddressOf scheduler_InitAppointmentDisplayText
			End If
		End Sub
		Protected Overrides Overloads Sub OnDeactivating()
			MyBase.OnDeactivating()
			If listEditor IsNot Nothing Then
				Dim scheduler As SchedulerControl = CType(listEditor.Scheduler, SchedulerControl)
				RemoveHandler scheduler.InitAppointmentDisplayText, AddressOf scheduler_InitAppointmentDisplayText
			End If
		End Sub
		Private Sub scheduler_InitAppointmentDisplayText(ByVal sender As Object, ByVal e As AppointmentDisplayTextEventArgs)
			Dim appointment As Appointment = e.Appointment
			Dim obj As DevExpress.Persistent.BaseImpl.Event = CType(listEditor.SourceObjectHelper.GetSourceObject(appointment), DevExpress.Persistent.BaseImpl.Event)
			e.Text = String.Format("{0}-({1})", e.Text, obj.GetHashCode())
		End Sub
	End Class
End Namespace